sources/networkmanager/global-network.go

```diff
@@ -0,0 +1,117 @@
+		Scope:           scope,
+		Attributes:      attributes,
+		Tags:            tagsToMap(gn.Tags),
+		LinkedItemQueries: []*sdp.LinkedItemQuery{
+			{
+				Query: &sdp.Query{
+					// +overmind:link networkmanager-site
+					// Search for all sites with this global network
+					Type:   "networkmanager-site",
+					Method: sdp.QueryMethod_SEARCH,
+					Query:  *gn.GlobalNetworkId,
+					Scope:  scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// ?? Sites can affect the global network
+					In: true,
+					// The global network will definitely affect the site
+					Out: true,
+				},
+			},
+		},
+	}
+
+	switch gn.State {
+	case types.GlobalNetworkStatePending:
+		item.Health = sdp.Health_HEALTH_PENDING.Enum()
+	case types.GlobalNetworkStateAvailable:
+		item.Health = sdp.Health_HEALTH_OK.Enum()
+	case types.GlobalNetworkStateUpdating:
+		item.Health = sdp.Health_HEALTH_PENDING.Enum()
+	case types.GlobalNetworkStateDeleting:
+		item.Health = sdp.Health_HEALTH_PENDING.Enum()
+	}
+
+	return &item, nil
+}
+
+//go:generate docgen ../../docs-data
+// +overmind:type networkmanager-global-network
+// +overmind:descriptiveType Netwotkmanager Global Network
+// +overmind:get Get a global network by id
+// +overmind:list List all global networks
+// +overmind:search Search for a global network by ARN
+// +overmind:group AWS
+// +overmind:terraform:queryMap aws_networkmanager_global_network.arn
+// +overmind:terraform:method SEARCH
+
+func NewGlobalNetworkSource(config aws.Config, accountID string, region string) *sources.AlwaysGetSource[*networkmanager.DescribeGlobalNetworksInput, *networkmanager.DescribeGlobalNetworksOutput, *networkmanager.DescribeGlobalNetworksInput, *networkmanager.DescribeGlobalNetworksOutput, NetworkmanagerClient, *networkmanager.Options] {
+	return &sources.AlwaysGetSource[*networkmanager.DescribeGlobalNetworksInput, *networkmanager.DescribeGlobalNetworksOutput, *networkmanager.DescribeGlobalNetworksInput, *networkmanager.DescribeGlobalNetworksOutput, NetworkmanagerClient, *networkmanager.Options]{
+		ItemType:  "networkmanager-global-network",
+		Client:    networkmanager.NewFromConfig(config),
+		AccountID: accountID,
+		Region:    region,
+		GetFunc:   globalNetworkGetFunc,
+		GetInputMapper: func(scope, query string) *networkmanager.DescribeGlobalNetworksInput {
+			return &networkmanager.DescribeGlobalNetworksInput{
+				GlobalNetworkIds: []string{
+					query,
+				},
+			}
+		},
+		ListInput: &networkmanager.DescribeGlobalNetworksInput{},
+		ListFuncPaginatorBuilder: func(client NetworkmanagerClient, input *networkmanager.DescribeGlobalNetworksInput) sources.Paginator[*networkmanager.DescribeGlobalNetworksOutput, *networkmanager.Options] {
+			return networkmanager.NewDescribeGlobalNetworksPaginator(client, input)
+		},
+		ListFuncOutputMapper: func(output *networkmanager.DescribeGlobalNetworksOutput, input *networkmanager.DescribeGlobalNetworksInput) ([]*networkmanager.DescribeGlobalNetworksInput, error) {
```

Comment:
It looks to me like the resaponse here returns full objects i.e. `[]types.GlobalNetwork`. an `AlwaysGetSource` is designed for secnarios where running a list doesn't give you all the data, for instance some APIs will just list the names of the resources, then you need to do a get for each of those names in order to get the data. If this API is truly able to return the whole object for both, then it should be converted to a `DescribeOnlySource` which will mean it won't make a Get for every response in the List, it'll simply convert the List response directly. You'll have to rework the `GetFunc` into an `OutputMapper` but it shouldn't be much work. See the `ec2` folder for lots of examples of this.